test/unit/clients/connector_client/connector_patch_apple_pay_toggle_test.js

```diff
@@ -0,0 +1,94 @@
+const path = require('path')
+const PactInteractionBuilder = require('../../../fixtures/pact_interaction_builder').PactInteractionBuilder
+const Connector = require('../../../../app/services/clients/connector_client').ConnectorClient
+const gatewayAccountFixtures = require('../../../fixtures/gateway_account_fixtures')
+
+// Constants
+const ACCOUNTS_RESOURCE = '/v1/api/accounts'
+const port = Math.floor(Math.random() * 48127) + 1024
+const connectorClient = new Connector(`http://localhost:${port}`)
+const expect = chai.expect
+const existingGatewayAccountId = 666
+
+// Global setup
+chai.use(chaiAsPromised)
+
+describe('connector client - patch apple pay toggle (enabled) request', () => {
+  const params = { path: 'allow_apple_pay', value: true }
+  const request = gatewayAccountFixtures.validGatewayAccountWalletToggleRequest(params).getPlain()
+
+  let provider = Pact({
+    consumer: 'selfservice',
+    provider: 'connector',
+    port: port,
+    log: path.resolve(process.cwd(), 'logs', 'mockserver-integration.log'),
+    dir: path.resolve(process.cwd(), 'pacts'),
+    spec: 2,
+    pactfileWriteMode: 'merge'
+  })
+
+  before(() => provider.setup())
+  after(() => provider.finalize())
+
+  describe('apple pay toggle - supported payment provider request', () => {
+    const applePayToggleSupportedPaymentProviderState =
+      `a gateway account supporting digital wallet with external id ${existingGatewayAccountId} exists in the database`
+
+    before(() => {
+      return provider.addInteraction(
+        new PactInteractionBuilder(`${ACCOUNTS_RESOURCE}/${existingGatewayAccountId}`)
+          .withUponReceiving('a valid patch apple pay toggle (enabled) request')
+          .withState(applePayToggleSupportedPaymentProviderState)
+          .withMethod('PATCH')
+          .withRequestBody(request)
+          .withStatusCode(200)
+          .withResponseHeaders({})
+          .build())
+    })
+
+    afterEach(() => provider.verify())
+
+    it('should toggle successfully', done => {
+      connectorClient.toggleApplePay(existingGatewayAccountId, true, null)
+        .should.be.fulfilled
+        .notify(done)
+    })
+  })
+
+  describe('apple pay toggle with unsupported payment provider request', () => {
+    const applePayToggleUnsupportedPaymentProviderState = `User ${existingGatewayAccountId} exists in the database`
+    const unsupportedPaymentProvider = 'sandbox'
+    const errorResponse = {
+      message: `Gateway ${unsupportedPaymentProvider} does not support digital wallets.`
+    }
+
+    before(() => {
+      return provider.addInteraction(
+        new PactInteractionBuilder(`${ACCOUNTS_RESOURCE}/${existingGatewayAccountId}`)
+          .withUponReceiving('a valid patch apple pay toggle (enabled) request')
+          .withState(applePayToggleUnsupportedPaymentProviderState)
+          .withMethod('PATCH')
+          .withRequestBody(request)
+          .withStatusCode(400)
+          .withResponseBody(errorResponse)
```

Comment:
I think since we don't use the response body, e.g. the error message then this can be removed `withResponseBody(errorResponse)`. My understanding is that this will create a pact which the pact broker will run against Connector and make sure that Connector's response includes the expected error message. Since Selfservice doesn't rely upon it, then I think the point @stephencdaly is making is that we shouldn't create a pact between SelfService and Connector that enforces it.

Reply:
I understand. I guess the point here was to make it obvious that **sandbox** doesn't support digital wallets, but if you think it's clear without it, then I can remove it.